src/pool-cl/CLPositionManager.sol

```diff
@@ -428,25 +359,74 @@ contract CLPositionManager is
         }
     }
 
+    /// @notice an internal helper used by CLNotifier
+    function _setSubscribed(uint256 tokenId) internal override {
+        positionInfo[tokenId] = positionInfo[tokenId].setSubscribe();
+    }
+
+    /// @notice an internal helper used by CLNotifier
+    function _setUnsubscribed(uint256 tokenId) internal override {
+        positionInfo[tokenId] = positionInfo[tokenId].setUnsubscribe();
+    }
+
     /// @dev overrides solmate transferFrom in case a notification to subscribers is needed
     function transferFrom(address from, address to, uint256 id) public virtual override {
         super.transferFrom(from, to, id);
-        if (positionConfigs[id].hasSubscriber()) _notifyTransfer(id, from, to);
+        if (positionInfo[id].hasSubscriber()) _notifyTransfer(id, from, to);
     }
 
     /// @inheritdoc ICLPositionManager
-    function getPositionLiquidity(uint256 tokenId, PositionConfig calldata config)
+    function getPoolAndPositionInfo(uint256 tokenId)
         public
         view
-        returns (uint128 liquidity)
+        returns (PoolKey memory poolKey, CLPositionInfo info)
     {
-        liquidity = clPoolManager.getLiquidity(
-            config.poolKey.toId(), address(this), config.tickLower, config.tickUpper, bytes32(tokenId)
-        );
+        info = positionInfo[tokenId];
+        poolKey = poolKeys[info.poolId()];
+    }
+
+    /// @inheritdoc ICLPositionManager
+    function getPositionLiquidity(uint256 tokenId) external view returns (uint128 liquidity) {
+        (PoolKey memory poolKey, CLPositionInfo info) = getPoolAndPositionInfo(tokenId);
+        return _getLiquidity(tokenId, poolKey, info.tickLower(), info.tickUpper());
     }
 
     /// @inheritdoc ICLPositionManager
-    function getPositionConfigId(uint256 tokenId) external view returns (bytes32) {
-        return positionConfigs[tokenId].getConfigId();
+    function positions(uint256 tokenId)
```

Comment:
i wonder if it might be possible to also include `hasSubscriber` in this `positions(uint256 tokenId)` -- all the other fields seems to be here 

in the end maybe FE will just call `positions(uint256 tokenId)` most of the time